contents/df_end.md

```diff
@@ -0,0 +1,224 @@
+## Index and Summarize
+
+```{=comment}
+summaries
+```
+
+Let's go back to the example data defined above
+
+```jl
+sco("""
+without_caption_label( # hide
+grades_2020()
+) # hide
+""")
+```
+
+To get a the **vector** for `name` back, we can use
+
+```{=comment}
+These two functions cannot be replaced by inline code due to
+`df.name` being converted to the same filename as `df[!, :name]` in Books.jl.
+I need to fix it.
+```
+
+```jl
+@sco JDS.names_grades1()
+```
+
+or
+
+```jl
+@sco JDS.names_grades2()
+```
+
+For the **row**, say the second row, we can use
+
+```jl
+sco("""
+df[2, :]
+df = DataFrame(df[2, :]) # hide
+without_caption_label(df) # hide
+""")
+```
+
+or create a function to give us row `i`
+
+```jl
+@sc JDS.grade_2020(1)
+```
+
+```jl
+sco("""
+without_caption_label( # hide
+grade_2020(2)
+) # hide
+""")
+```
+
+Continuing on this, we can also get only `names` for the first 2 rows:
+
+```jl
+@sco JDS.grades_indexing()
+```
+
+If we assume that all names in the table are unique, we can also write a function to obtain the grade for a person via their `name`.
+To do this, we convert the table back to one of Julia's basic data structures which is capable of creating a mappings, namely dictionaries:
+
+```jl
+@sc grade_2020("")
+```
+
+```jl
+sco("""
+grade_2020("Bob")
+""")
+```
+
+which works because `zip` loops through `df.name` and `df.grade_2020` at the same time like a zipper:
```

Comment:
```suggestion
which works because `zip()` loops through `df.name` and `df.grade_2020` at the same time like a zipper:
```

Reply:
Is the idea that `f()` denotes a function which can take any parameters a common idea? I could understand that it is confusing for the reader, since one would think that `f` takes no arguments.

How about `zip(...)` and `DataFrame(...)` instead?

Reply:
We could do that. I want to make a distinction on `variables`, `functions(...)`. `Packages.jl`, `Types` and `Constructors(...)`.
If you define `function(...)` please let me know so I can standardize it in all the sections that I wrote. I've used the `function()` without the `...`